modules/light-clients/07-tendermint/types/update_test.go

```diff
@@ -35,28 +38,19 @@ func (suite *TendermintTestSuite) TestCheckHeaderAndUpdateState() {
 	heightPlus5 := clienttypes.NewHeight(height.RevisionNumber, height.RevisionHeight+5)
 
 	altVal := tmtypes.NewValidator(altPubKey, revisionHeight)
-
-	// Create bothValSet with both suite validator and altVal. Would be valid update
-	bothValSet := tmtypes.NewValidatorSet(append(suite.valSet.Validators, altVal))
 	// Create alternative validator set with only altVal, invalid update (too much change in valSet)
 	altValSet := tmtypes.NewValidatorSet([]*tmtypes.Validator{altVal})
 
-	signers := []tmtypes.PrivValidator{suite.privVal}
-
-	// Create signer array and ensure it is in same order as bothValSet
-	_, suiteVal := suite.valSet.GetByIndex(0)
-	bothSigners := ibctesting.CreateSortedSignerArray(altPrivVal, suite.privVal, altVal, suiteVal)
-
 	altSigners := []tmtypes.PrivValidator{altPrivVal}
 
 	testCases := []struct {
 		name    string
-		setup   func()
+		setup   func(suite *TendermintTestSuite)
 		expPass bool
 	}{
 		{
 			name: "successful update with next height and same validator set",
-			setup: func() {
+			setup: func(suite *TendermintTestSuite) {
```

Comment:
what was the purpose of this? I'm pretty sure all the setup funcs have access to suite regardless

Reply:
Need to reset the suite on each case to avoid metadata writes messing things up. Have to pass in suite so that the function uses the current suite's signer values rather than original suite (which will be different)